rust/stub/src/pe_loader.rs

```diff
@@ -179,12 +178,10 @@ impl Image {
         let status = (self.entry)(handle, unsafe { system_table.unsafe_clone() });
 
         // If the kernel entry point returned, deallocate its image, and restore our loaded image handle.
-        // If it calls Exit(), then we don't get here, and our resources are leaked.
-        // I don't see any way to avoid this.
+        // If the kernel exits boot services, we don't get here and there is no point in freeing memory.
```

Comment:
This should at most be an addition instead of a replacement. I don't mean the kernel calling `ExitBootServices` and then (hopefully) not returning. I really mean `Exit`, which normally returns to the parent. However, since we reuse our image handle for the kernel, it actually returns to systemd-boot, and our resources are leaked.

Reply:
Ah good point. Feel free to fix!